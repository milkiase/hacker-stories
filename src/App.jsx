import * as React from 'react'

const initialStories = [
  {
  title: 'React',
  url: 'https://reactjs.org/',
  author: 'Jordan Walke',
  num_comments: 3,
  points: 4,
  objectID: 0,
  },
  {
  title: 'Redux',
  url: 'https://redux.js.org/',
  author: 'Dan Abramov, Andrew Clark',
  num_comments: 2,
  points: 5,
  objectID: 1,
},
]
const [storiesFetchInit, storiesFetchSuccess, storiesFetchFail, removeStory] = ['STORIES_FETCH_INIT', 'STORIES_FETCH_SUCCESS', 'STORIES_FETCH_FAIL', 'REMOVE_STORY']
const storiesReducer = (state, action)=>{
  switch (action.type){
    case storiesFetchInit:
      return {...state, isLoading: true, isError: false}
    case storiesFetchSuccess:
      return {...state, isLoading: false, isError: false, data: action.payload}
    case storiesFetchFail:
      return {...state, isLoading: false, isError: true}
    case removeStory:
      return {...state, isLoading: false, isError: false, data: state.data.filter((story)=>story.objectID != action.payload)}
    default:
      throw new Error()
  }
}
const App = ()=>{
  const [stories, dispatchStories] = React.useReducer(storiesReducer, {data: [], isLoading: false, isError: false});
  // const [isLoading, setIsLoading] = React.useState(false)
  // const [thereIsError, setThereIsError] = React.useState(false)
  const getAsyncStories = new Promise((resolve, reject)=>{
    setTimeout(()=>{
      resolve({data: {
        stories: initialStories
      }})
      reject()
    }, 2000)
  })
  React.useEffect(()=>{
    // setIsLoading(true)
    dispatchStories({
      type: storiesFetchInit
    })
    getAsyncStories.then((response)=>{
      dispatchStories({
        type : storiesFetchSuccess,
        payload: response.data.stories
      })
      // setIsLoading(false)
    }).catch(()=>{
      dispatchStories({
        type: storiesFetchFail
      })
      // setThereIsError(true)
      // setIsLoading(false)
    })
  }, [])
  const useStateStorage = (key, initialState)=>{
    const [value, setValue] = React.useState(localStorage.getItem(key) ?? initialState)
    React.useEffect(()=>{
      localStorage.setItem(key, value)
    }, [value, key])
    return [value, setValue]
  }
  const [searchTerm, searchTermHandler] = useStateStorage('seachTerm', 'React')
  const searchedStories = stories.data.filter((story)=>story.title.toLowerCase().includes(searchTerm.toLowerCase()))
  
  const handleSearch = (event)=>{
    searchTermHandler(event.target.value)
  }
  const resetSearchTerm = ()=>{
    searchTermHandler('react')
  }
  const handleRemoveStory = (_id)=>{
    dispatchStories({
      type: removeStory,
      payload: _id
    })
  }
  
  return <>
            <InputWithText id='search'  onValueChange={handleSearch} value={searchTerm} autoFocus>
              <strong>Search :</strong>
            </InputWithText>
            <BaseBtn onClick={resetSearchTerm} classValue=' bg-teal-600 rounded-sm ml-3 px-2 py-1'>Reset</BaseBtn>
            {stories.isError && <p className=' text-red-800'>OOops.. there have been some server/network error.</p>}
            {stories.isLoading ? <span className="relative block">
                          <div type="div" className="inline-flex items-center px-4 py-2 font-semibold leading-6 border ml-16 mt-4 rounded-md text-sky-500 bg-white transition ease-in-out duration-150 cursor-not-allowed ring-1 ring-slate-900/10 dark:ring-slate-200/20" >
                            Loading stories...
                          </div>
                          <span className="flex absolute h-3 w-3 top-0 right-0 mt-3 -mr-1">
                            <span className="animate-ping absolute inline-flex h-full w-full rounded-full bg-sky-400 opacity-75"></span>
                            <span className="relative inline-flex rounded-full h-3 w-3 bg-sky-500"></span>
                          </span>
                        </span> : <List list={searchedStories} onItemRemove={handleRemoveStory}/>}
        
      </>
}

const List = ({list, onItemRemove})=>{
  return (
    <ul>
      {
        list.map((element)=>  <Item key={element.objectID} {...element} onRemoveSelf={onItemRemove}></Item>)
      }
    </ul>
  );
}

const Item = ({url, title, author, num_comments, points, objectID, onRemoveSelf})=>
    {
      return (<div>
      <li className=" ">
        <span>
          <a className=" text-blue-700" href={url} target="_blank" rel="noopener noreferrer">{title}</a>
          <BaseBtn classValue="px-2 mt-1 mx-3 bg-red-500 hover:bg-red-600 text-white" onClick={()=>onRemoveSelf(objectID)}>X</BaseBtn>
        </span>
        <br />Authors: <span>{author} </span>
        <br />Comments: <span> {num_comments} </span>
        <br />Points: <span> {points} </span>
      </li>
      <hr />
    </div>)}

const InputWithText = ({id, children, type = 'text', value, onValueChange, autoFocus})=>{
  const inputRef = React.useRef()
  React.useEffect(()=>{
    if(autoFocus && inputRef.current){
      inputRef.current.focus()
    }
  }, [autoFocus])
  return (
    <>
      <label htmlFor={id}>{children} </label>
      <input onChange={onValueChange} type={type} value={value} id={id} ref={inputRef}
              className=" px-2 py-1 border rounded-sm border-gray-950"/>
    </>
  );
}
const BaseBtn = ({children, onClick, classValue})=>{
  return (
    <button type="button" onClick={onClick} className={classValue}>{children}</button>
  )
}
export default App
