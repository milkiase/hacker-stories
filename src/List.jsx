import BaseBtn from "./BaseBtn";

const List = ({list, onItemRemove})=>{
    return (
      <ul>
        {
          list.map((element)=>  <Item key={element.objectID} {...element} onRemoveSelf={onItemRemove}></Item>)
        }
      </ul>
    );
  }

  const Item = ({url, title, author, num_comments, points, objectID, onRemoveSelf})=>
      {
        return (<div>
        <li className=" ">
          <span>
            <a className=" text-blue-700" href={url} target="_blank" rel="noopener noreferrer">{title}</a>
            <BaseBtn classValue="px-2 mt-1 mx-3 bg-red-500 hover:bg-red-600 text-white" onClick={()=>onRemoveSelf(objectID)}>X</BaseBtn>
          </span>
          <br />Authors: <span>{author} </span>
          <br />Comments: <span> {num_comments} </span>
          <br />Points: <span> {points} </span>
        </li>
        <hr />
      </div>)}

export default List;
export {Item}