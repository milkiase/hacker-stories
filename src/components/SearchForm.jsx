import InputWithText from "./InputWithText";
import BaseBtn from "./BaseBtn";
const SearchForm = ({onSearchSubmit, onSearchInput, searchTerm})=>{
    return (
      <form onSubmit={onSearchSubmit} className='flex'>
        <InputWithText id='search'  onValueChange={onSearchInput} value={searchTerm} autoFocus>
          <strong>Search :</strong>
        </InputWithText>
        <BaseBtn type='submit' disabled={!searchTerm}
          classValue=' bg-teal-600 text-white rounded-sm ml-3 px-2 py-1 mt-6 mr-10 md:mt-0 md:mr-0'>Submit</BaseBtn>
      </form>
    );
  }

export default SearchForm