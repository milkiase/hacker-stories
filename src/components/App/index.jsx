/* eslint-disable react-refresh/only-export-components */
import * as React from 'react'
import axios from 'axios'
import { sortBy } from 'lodash'

import List from '../List'
import SearchForm from '../SearchForm'
import Options from "../Options";
import Switch from '../Switch';
import SearchHistory from '../../SearchHistory'

const API_ENDPOINT = 'https://hn.algolia.com/api/v1/search?query='
const SORT_KEY_MAPPING = {
  None: '',
  Title: 'title',
  Author: 'author',
  Comments: 'num_comments',
  Points: 'points'
}

const [storiesFetchInit, storiesFetchSuccess, storiesFetchFail, removeStory] = ['STORIES_FETCH_INIT', 'STORIES_FETCH_SUCCESS', 'STORIES_FETCH_FAIL', 'REMOVE_STORY']
const storiesReducer = (state, action)=>{

  switch (action.type){
    case storiesFetchInit:
      return {...state, isLoading: true, isError: false}
    case storiesFetchSuccess:
      return {...state, isLoading: false, isError: false, isSorted: false, data: action.payload}
    case storiesFetchFail:
      return {...state, isLoading: false, isError: true}
    case removeStory:
      return {...state, isLoading: false, isError: false, data: state.data.filter((story)=>story.objectID != action.payload)}
    case 'SORT_STORIES':
      if((state.data.length == 0) || (action.payload.sortBy == 'None')){
        return {...state, isSorted: true}
      }
      if(isNaN(state.data[0][SORT_KEY_MAPPING[action.payload.sortBy]])){
        if(action.payload.sortInDescending){
          return { ...state, isSorted: true, data: sortBy(state.data, SORT_KEY_MAPPING[action.payload.sortBy])}
        }else{
          return {...state, isSorted: true, data: sortBy(state.data, SORT_KEY_MAPPING[action.payload.sortBy]).reverse()}
        }
      }else{
        if(action.payload.sortInDescending){
          return { ...state, isSorted: true, data: sortBy(state.data, SORT_KEY_MAPPING[action.payload.sortBy]).reverse()}
        }else{
          return {...state, isSorted: true, data: sortBy(state.data, SORT_KEY_MAPPING[action.payload.sortBy])}
        }
      }
      
    default:
      throw new Error()
  }
}
const getUrl = (searchTerm)=>`${API_ENDPOINT}${searchTerm}`
const getActiveUrl = (urls)=>urls[urls.length - 1]
const App = ()=>{
  const [stories, dispatchStories] = React.useReducer(storiesReducer, {data: [], isLoading: false, isError: false, isSorted: false});
  const useStateStorage = (key, initialState)=>{
    const [value, setValue] = React.useState(localStorage.getItem(key) ?? initialState)
    React.useEffect(()=>{
      localStorage.setItem(key, value)
    }, [value, key])
    return [value, setValue]
  }
  const [searchTerm, searchTermHandler] = useStateStorage('seachTerm', 'React')
  const [urls, setUrls] = React.useState([getUrl(searchTerm)])
  const [sort, setSort] = React.useState({sortBy: 'None', isDescending: true})

  const extractSearchTerm = (url)=> url.replace(API_ENDPOINT, '')
  const getRecentSearches = ()=>{
    let recent = urls.slice(-6, -1)
    return recent.map(extractSearchTerm).reverse()
  }
  const handleFetchStories = React.useCallback(async ()=>{
    dispatchStories({
      type: storiesFetchInit
    })
    try {
      const result = await axios.get(getActiveUrl(urls))
      dispatchStories({
        type : storiesFetchSuccess,
        payload: result.data.hits
      })
    } catch{
      dispatchStories({
        type: storiesFetchFail
      })
    }
    
  }, [urls])
  
  const handleSortStory = React.useCallback(()=>{
    dispatchStories({
      type: 'SORT_STORIES',
      payload: {
        sortBy: sort.sortBy,
        sortInDescending: sort.isDescending
      }
    })
  }, [sort.isDescending, sort.sortBy])

  React.useEffect(()=>{
    handleFetchStories()
  }, [handleFetchStories])
  
  const handleSearchInput = (event)=>{
    searchTermHandler(event.target.value)
  }
  const handleSearch = (event, value)=>{
    event.preventDefault();    
    let newUrl = getUrl(value)
    if(!value || (newUrl == getActiveUrl(urls))) return;
    setUrls(urls.concat(newUrl));
  }
  const handleSearchSubmit = (event)=>{
    handleSearch(event, searchTerm)
  }
  const handleRemoveStory = (_id)=>{
    dispatchStories({
      type: removeStory,
      payload: _id
    })
  }

  React.useEffect(()=>{
    if(!stories.isSorted){
      handleSortStory()
    }
  }, [stories, handleSortStory])

  React.useEffect(()=>{
    handleSortStory()
  }, [handleSortStory])

  const handleHistoryClick = (event)=>{
    searchTermHandler(event.target.value)
    handleSearch(event, event.target.value)
  }
  
  return <div className='p-3'>
            <div className="flex flex-col gap-5 pb-4 border-b">
            <div className="flex flex-wrap gap-1">
              <SearchForm onSearchSubmit={handleSearchSubmit} onSearchInput={handleSearchInput} searchTerm={searchTerm} />
              <SearchHistory searchHistory={getRecentSearches()} onHistoryClicked={handleHistoryClick}></SearchHistory>
            </div>
              <div className="flex align-top text-xs">
                <Options options={['None', 'Title', 'Author', 'Comments', 'Points']} onOptionChange={(value)=>{setSort({sortBy: value, isDescending: sort.isDescending})}}></Options>
                {(sort.sortBy != 'None') && <Switch switchOn={sort.isDescending} onSwitchToggle={()=>{ setSort({sortBy: sort.sortBy, isDescending: !sort.isDescending}) }}></Switch>}
              </div>
            </div>
            {stories.isError && <p className=' text-red-800'>OOops.. there have been some server/network error.</p>}
            {stories.isLoading ? <span className="relative block">
                          <div type="div" className="inline-flex items-center px-4 py-2 font-semibold leading-6 border ml-16 mt-4 rounded-md text-sky-500 bg-white transition ease-in-out duration-150 cursor-not-allowed ring-1 ring-slate-900/10 dark:ring-slate-200/20" >
                            Loading stories...
                          </div>
                          <span className="flex absolute h-3 w-3 top-0 right-0 mt-3 -mr-1">
                            <span className="animate-ping absolute inline-flex h-full w-full rounded-full bg-sky-400 opacity-75"></span>
                            <span className="relative inline-flex rounded-full h-3 w-3 bg-sky-500"></span>
                          </span>
                        </span> : (
                          <List list={stories.data} onItemRemove={handleRemoveStory}/>
                        )}
        
      </div>
}


export default App
export {storiesReducer}