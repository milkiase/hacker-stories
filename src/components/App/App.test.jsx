import {describe, it, expect, vi} from 'vitest'
import App, { storiesReducer } from "./index.jsx";
import SearchForm from '../../components/SearchForm.jsx';
import List, {Item} from '../../components/List.jsx';
import {
    render,
    screen,
    fireEvent,
    waitFor
} from '@testing-library/react'
import axios from 'axios';
vi.mock('axios')
const storyOne = {
    title: 'React',
    url: 'https://reactjs.org/',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
}
const storyTwo = {
    title: 'Redux',
    url: 'https://redux.js.org/',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 1,
}
const stories = [storyOne, storyTwo]

describe('storiesReducer', ()=>{
    it('removes a story from all the stories', ()=>{
        const state = {
            data : stories,
            isLoading: false,
            isError: false
        }
        const action = {
            type: 'REMOVE_STORY',
            payload: storyTwo.objectID
        }
        const newState = storiesReducer(state, action)
        const expectedState = {
            data: [storyOne],
            isLoading: false,
            isError: false
        }
        expect(newState).toStrictEqual(expectedState)
    });
    it('set isLoading to true on FetchInit', ()=>{
        const state = {
            data: [],
            isLoading: false,
            isError: true
        }
        const action = {
            type: 'STORIES_FETCH_INIT'
        }
        const newState = storiesReducer(state, action)
        const expectedState = {
            data: [],
            isError: false,
            isLoading: true
        }
        expect(newState).toStrictEqual(expectedState)
    });
    it('fetch success test', ()=>{
        const state = {
            data: [],
            isLoading: true,
            isError: true
        }
        const action = {
            type: 'STORIES_FETCH_SUCCESS',
            payload: stories
        }
        const newState = storiesReducer(state, action)
        const expectedState = {
            data: stories,
            isError: false,
            isLoading: false
        }
        expect(newState).toStrictEqual(expectedState)
    });
    it('fetch stories Failed', ()=>{
        const state = {
            data: [],
            isLoading: true,
            isError: false
        }
        const action = {
            type: 'STORIES_FETCH_FAIL'
        }
        const expectedState = {
            data: [],
            isLoading: false,
            isError: true
        }
        const newState = storiesReducer(state, action)
        expect(newState).toStrictEqual(expectedState)
    })
})

describe('Item', ()=>{
    it('renders all properties', ()=>{
        render(<Item {...storyOne}></Item>);
        // screen.debug()
        expect(screen.getByText('Jordan Walke')).toBeInTheDocument();
        expect(screen.getByText('React')).toHaveAttribute('href', 'https://reactjs.org/')
    })
    it('renders a clickble dismiss button', ()=>{
        render(<Item {...storyOne}/>)
        expect(screen.getByRole('button')).toBeInTheDocument()
    });
    it('remove handler function is being called', ()=>{
        const handleRemoveItem = vi.fn();
        render(<Item {...storyOne} onRemoveSelf={handleRemoveItem}/>);
        fireEvent.click(screen.getByRole('button'));
        expect(handleRemoveItem).toHaveBeenCalledTimes(1);
    })
})

describe('SearchForm', ()=>{
    const searchFormProps = {
        searchTerm : 'React',
        onSearchSubmit: vi.fn(),
        onSearchInput: vi.fn()
    }
    it('it renders the input field with its value', ()=>{
        render(<SearchForm {...searchFormProps}/>)
        // screen.debug()
        expect(screen.getByDisplayValue('React')).toBeInTheDocument()
    });
    it('renders the correct label', ()=>{
        render(<SearchForm {...searchFormProps}></SearchForm>)
        expect(screen.getByLabelText(/Search/)).toBeInTheDocument()
    });
    it('calls onSearchInput on input change', ()=>{
        render(<SearchForm {...searchFormProps}></SearchForm>)
        fireEvent.change(screen.getByDisplayValue('React'), {target: {value: 'Redux'}})
        expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1)
    });
    it('calls onSeachSubmit on button click', ()=>{
        render(<SearchForm {...searchFormProps} />)
        fireEvent.click(screen.getByRole('button'))
        expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1)
        // expect(searchFormProps.onSearchSubmit).toHaveBeenCalledWith('React')
    });
    it('renders snapshots', ()=>{
        const {container} = render(<SearchForm {...searchFormProps} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})

describe('List', ()=>{
    it('renders all items', ()=>{
        render(<List list={stories}></List>)
        expect(screen.getAllByText(/Comments/)[1]).toBeInTheDocument()
    });
    it('renders snapshot', ()=>{
        const {container} = render(<List list={stories}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
});

describe('App', ()=>{
    it('succeeds fetching data', async ()=>{
        const promise = Promise.resolve({data: {hits: stories}})

        axios.get.mockImplementationOnce(()=> promise);
        render(<App></App>)
        // screen.debug()
        expect(screen.getByText(/Loading/)).toBeInTheDocument();
        await waitFor(async ()=>await promise);
        // screen.debug()
        expect(screen.queryByText(/Loading/)).toBeNull()
        expect(screen.getByText('React')).toBeInTheDocument()
        expect(screen.getByText('Redux')).toBeInTheDocument()
        expect(screen.getAllByText(/Comments:/).length).toBe(2)
    });
    
    it('fails fetching data', async ()=>{
        const promise = Promise.reject()
        axios.get.mockImplementationOnce(()=> promise)
        render(<App></App>)
        expect(screen.getByText(/Loading/)).toBeInTheDocument()
        try{
            await waitFor(async()=>await promise)
        }catch(error){
            expect(screen.queryByText(/Loading/)).toBeNull()
            expect(screen.getByText(/error/)).toBeInTheDocument()
        }
    });

    it('removes an item when dismiss button is clicked', async ()=>{
        const promise = Promise.resolve({data: {hits: stories}})
        axios.get.mockImplementationOnce(()=> promise)

        render(<App></App>)
        expect(screen.getByText(/Loading/)).toBeInTheDocument();
        await waitFor(async()=> await promise);
        expect(screen.queryByText(/Loading/)).toBeNull()
        expect(screen.queryByText('OOops.. there have been some server/network error.')).toBeNull()

        expect(screen.getByText(/Jordan Walke/)).toBeInTheDocument()
        fireEvent.click(screen.getAllByRole('button')[1])
        expect(screen.queryByText(/Jordan Walke/)).toBeNull()
    });
    it('searches for specific stories', async ()=>{
        const anotherStory = {
            title: 'JavaScript',
            url: 'https://en.wikipedia.org/wiki/JavaScript',
            author: 'Brendan Eich',
            num_comments: 15,
            points: 10,
            objectID: 3,
            };
        const reactPromise = Promise.resolve({data: {hits: stories}})
        const jsPromise = Promise.resolve({data: {hits: [anotherStory]}})
        axios.get.mockImplementation((url)=>{
            if(url.includes('React')){
                return reactPromise
            }else if(url.includes('Javascript')){
                return jsPromise
            }else{
                throw Error
            }
        })
        render(<App></App>)
        expect(screen.getByText(/Loading/)).toBeInTheDocument()
        await waitFor(async ()=> await reactPromise)
        expect(screen.getByText('React')).toBeInTheDocument()
        expect(screen.queryByText(/Loading/)).toBeNull()
        expect(screen.getByText('Jordan Walke')).toBeInTheDocument()
        expect(screen.getByText('Dan Abramov, Andrew Clark')).toBeInTheDocument()

        expect(screen.getByDisplayValue('React')).toBeInTheDocument()
        fireEvent.change(screen.getByDisplayValue('React'), {target: {value: 'Javascript'}})
        expect(screen.getByDisplayValue('Javascript')).toBeInTheDocument()
        expect(screen.queryByDisplayValue('React')).toBeNull()
        expect(screen.queryByText('Brendan Eich')).toBeNull()

        fireEvent.submit(screen.getByText('Submit'))
        await waitFor(async()=>await jsPromise)
        expect(screen.queryByText('Jordan Walke')).toBeNull()
        expect(screen.queryByText('Dan Abramov, Andrew Clark')).toBeNull()
        expect(screen.getByText('Brendan Eich')).toBeInTheDocument()
    });
});
