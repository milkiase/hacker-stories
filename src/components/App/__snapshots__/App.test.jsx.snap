// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List > renders snapshot 1`] = `
<ul>
  <div>
    <li
      class=" "
    >
      <span>
        <a
          class=" text-blue-700"
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          React
        </a>
        <button
          class="px-2 mt-1 mx-3 bg-red-500 hover:bg-red-600 text-white"
          type="button"
        >
          X
        </button>
      </span>
      <br />
      Authors: 
      <span>
        Jordan Walke
         
      </span>
      <br />
      Comments: 
      <span>
         
        3
         
      </span>
      <br />
      Points: 
      <span>
         
        4
         
      </span>
    </li>
    <hr />
  </div>
  <div>
    <li
      class=" "
    >
      <span>
        <a
          class=" text-blue-700"
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Redux
        </a>
        <button
          class="px-2 mt-1 mx-3 bg-red-500 hover:bg-red-600 text-white"
          type="button"
        >
          X
        </button>
      </span>
      <br />
      Authors: 
      <span>
        Dan Abramov, Andrew Clark
         
      </span>
      <br />
      Comments: 
      <span>
         
        2
         
      </span>
      <br />
      Points: 
      <span>
         
        5
         
      </span>
    </li>
    <hr />
  </div>
</ul>
`;

exports[`SearchForm > renders snapshots 1`] = `
<form
  class="flex"
>
  <div>
    <label
      for="search"
    >
      <strong>
        Search :
      </strong>
       
    </label>
    <input
      class=" px-2 py-1 border rounded-sm border-gray-950"
      id="search"
      type="text"
      value="React"
    />
  </div>
  <button
    class=" bg-teal-600 text-white rounded-sm ml-3 px-2 py-1 mt-6 mr-10 md:mt-0 md:mr-0"
    type="submit"
  >
    Submit
  </button>
</form>
`;
